api/layer/layer.go

```diff
@@ -281,7 +282,7 @@ func (n *layer) EphemeralKey() *keys.PublicKey {
 }
 
 func (n *layer) Initialize(ctx context.Context, c Notificator) error {
-	if c == nil {
+	if c == nil || reflect.ValueOf(c).Kind() != reflect.Struct && reflect.ValueOf(c).IsNil() {
```

Comment:
Can we avoid reflection until it is absolutely necessary?
We can cast interface to exact structure and check nil value there, can't we?

Reply:
Then we need know exactly structure type/s. It seems that using an interface as a parameter makes no sense in this case

Reply:
Why we should avoid using of reflect?

Reply:
It's like a marker that our code does something unsafe and dirty though we actually don't do this. It also makes the code less readable.